### ISSUE TITLE
`success` events emitted for failed job

### ISSUE THREAD

**Issue by ghost** [hokify/agenda#48]
documentation states that `success` and `success:job name` events will be emitted only when a job succeeds, but if the job is failed manually by calling `job.fail(err).save();` the messages will be emitted anyway.

```js
const agenda = new Agenda();

const handler = (job) => job.fail('error').save();
agenda.define('test', handler);

agenda.on('success', () => console.log('success'));

await agenda.start();
await agenda.now('test');

// will print 'success' even if the job has failed
```

[link to code](https://github.com/hokify/agenda/blob/main/src/Job.ts#L431)

**Comment by riddlewiggler** [comment]
Shouldn't you first [disable][disable-url] the job @volpd?

It looks like [fail][fail-url] will only set some values of the specified job, it won't stop job execution. Correct me if I'm wrong.

[disable-url]: https://github.com/hokify/agenda#disable
[fail-url]: https://github.com/hokify/agenda/blob/main/src/Job.ts#L213

**Comment by ghost** [comment]
@riddlewiggler yes, i forgot that part on the example, still the problem is not that the job is executed.

what happens is:

- job is executed
- job is manually failed (and disabled) in the handler
- handler returns
- agenda emits `success` message, even if the job has failed

yes, i saw that `fail()` will only set the fail reason and update the `failCount` on the job, but I would have expected Agenda to check the status of the job before emitting `success` or `fail`, whereas it seems `fail` will be emitted if the handler throws an exception, and `success` will be emitted when the handler returns, regardless of the status of the job. 

I don't think this was mentioned in the documentation, if it is the desired behavior I think it should be clarified, since it was clear to me only after inspecting the code.
